storage/innobase/os/os0file.cc

```diff
@@ -3125,7 +3146,7 @@ AIO::linux_dispatch(Slot* slot, bool should_buffer)
 	ulint	io_ctx_index = slot->pos / slots_per_segment;
 
 	if (should_buffer) {
-		ut_ad(this == s_reads);
+		ut_ad(this == s_reads || this == s_ibuf);
 
 		acquire();
 		/* There are m_slots.size() elements in m_pending,
```

Comment:
Just one doubt: is it possible to buffer as many read request as possible? Or is there limit at AIO level? Like the m_slots defined here? (I think the read head is typically limited to an extent size which is 64 pages)